File: DayongManager/CollectionCycle.cs
using System;

namespace DayongManager;

public sealed class CollectionCycle
{
	public long Id { get; set; }

	public string Name { get; set; } = "";

	public string Type { get; set; } = "Dayong";

	public decimal ExpectedAmount { get; set; }

	public DateTime? DueDate { get; set; }

	public bool Active { get; set; } = true;
}

File: DayongManagerWeb/app/Models/CollectionCycle.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;

class CollectionCycle extends Model
{
    protected $guarded = [];
    protected function casts(): array { return ['expected_amount'=>'decimal:2','start_date'=>'date','due_date'=>'date','active'=>'boolean']; }
    public function payments(): HasMany { return $this->hasMany(Payment::class); }
}
